update naming for aggregation functions

brad-griffin opened the pull request:
update naming for aggregation functions across quantities of provided, requested, distributed, emissions classes and add agg functions

brad-griffin committed:
update naming for aggregation functions across quantities of provided, requested, distributed, emissions classes and add agg functions

jillianderson8 commented on CIMS/quantities.py:
```diff
@@ -47,63 +64,56 @@ def calculate_proportion(self, node, tech=None):
+                if service not in quantities[energy]:
+                    quantities[energy][service] = 0
+                quantities[energy][service] += self.requested_quantities[energy][service]
+        return quantities
```
I noticed the `sum_requested_by_energy_service()` method isn't being referenced anywhere in the code base. I'm guessing this is a user-facing function, rather than a forgotten but unneeded function? Can you confirm?

jillianderson8 replied:
Confirmed, this is a user-facing function. Will add a doc-string.

jillianderson8 commented on CIMS/quantities.py:
```diff
@@ -47,63 +64,56 @@ def calculate_proportion(self, node, tech=None):
+                if service not in quantities:
+                    quantities[service] = 0
+                quantities[service] += self.requested_quantities[energy][service]
+        return quantities
```
I noticed that the `sum_requested_by_service` method isn't being referenced anywhere in the code base. I'm guessing this is a user-facing function, rather than a forgotten but unneeded function? Can you confirm?

jillianderson8 replied:
Confirmed, this is a user-facing function. Will add a doc-string.

jillianderson8 commented on CIMS/emissions.py:
```diff
@@ -312,38 +312,157 @@ def __truediv__(self, other: int | float) -> Emissions:
         return result
 
-    def summarize(self) -> dict:
+    def sum_emissions_by_energy_ghg_type(self) -> dict:
```
I noticed that the `sum_emissions_by_energy_ghg_type()` method isn't being referenced anywhere in the code base. I'm guessing this is a user-facing function, rather than a forgotten but unneeded function? Can you confirm?

jillianderson8 replied:
Confirmed, this is a user-facing function. Will add a doc-string.

jillianderson8 commented on CIMS/emissions.py:
```diff
@@ -312,38 +312,157 @@ def __truediv__(self, other: int | float) -> Emissions:
+
+        return sum_emissions
+    
+    def sum_emissions_by_energy(self) -> dict:
```
I noticed that the `sum_emissions_by_energy()` method isn't being referenced anywhere in the code base. I'm guessing this is a user-facing function, rather than a forgotten but unneeded function? Can you confirm?

jillianderson8 replied:
Confirmed, this is a user-facing function. Will add a doc-string.

jillianderson8 commented on CIMS/emissions.py:
```diff
@@ -312,38 +312,157 @@ def __truediv__(self, other: int | float) -> Emissions:
+
+        return sum_emissions
+    
+    def sum_emissions_by_ghg(self) -> dict:
```
I noticed that the `sum_emissions_by_ghg()` method isn't being referenced anywhere in the code base. I'm guessing this is a user-facing function, rather than a forgotten but unneeded function? Can you confirm?

jillianderson8 replied:
Confirmed, this is a user-facing function. Will add a doc-string.

jillianderson8 commented on CIMS/emissions.py:
```diff
@@ -312,38 +312,157 @@ def __truediv__(self, other: int | float) -> Emissions:
+
+        return sum_emissions
+    
+    def sum_emissions_by_type(self) -> dict:
```
I noticed that the `sum_emissions_by_type()` method isn't being referenced anywhere in the code base. I'm guessing this is a user-facing function, rather than a forgotten but unneeded function? Can you confirm?

jillianderson8 replied:
Confirmed, this is a user-facing function. Will add a doc-string.

jillianderson8 commented on CIMS/emissions.py:
```diff
@@ -312,38 +312,157 @@ def __truediv__(self, other: int | float) -> Emissions:
+
+        return sum_emissions
+    
+    def sum_emissions_by_total(self) -> dict:
```
I noticed that the `sum_emissions_by_total()` method isn't being referenced anywhere in the code base. I'm guessing this is a user-facing function, rather than a forgotten but unneeded function? Can you confirm?

jillianderson8 replied:
Confirmed, this is a user-facing function. Will add a doc-string.

jillianderson8 commented on CIMS/emissions.py:
```diff
@@ -312,38 +312,157 @@ def __truediv__(self, other: int | float) -> Emissions:
+    emissions_by_child = emissions_child * provided_child_to_self / provided_child_total
 
-        return summary_emissions
+    return emissions_by_child
```
I believe `sum_emissions_by_service` is meant to be a method nested undr the `Emissions` class, rather than its own function. Please confirm. 

Note, this method isn't being referenced anywhere in the code base. I'm guessing this is a user-facing function, rather than a forgotten but unneeded function? Can you confirm?

jillianderson8 replied:
1. This is _not_ meant to be an `Emissions` class method. It is meant to be standalone. Adjust parameter names to make this more clear. (i.e. removed `self`)
2. Confirmed that this is a user-facing function. Will add a doc-string.

jillianderson8 commented on CIMS/quantities.py:
```diff
@@ -136,15 +146,18 @@ def record_distributed_supply(self, supply_node, distributed_supply_node, amount
+                supply[energy] += self.distributed_supply[energy][distributed_supply_node]
+        return supply
+
+    def sum_distributed_by_total(self):
```
I noticed the `sum_distributed_by_total()` method isn't being referenced anywhere in the code base. I'm guessing this is a user-facing function, rather than a forgotten but unneeded function? Can you confirm?

jillianderson8 replied:
Confirmed, this is a user-facing function. Will add a doc-string.

jillianderson8 reviewed (commented):
Pretty well everything looks good. 

- The only thing that might hinder functionality is that one function (`sum_emissions_by_service`) has been defined at the top level of the python file and isn't nested under the `Emissions` object. 
- There are a fair number of methods that aren't being referenced anywhere in the CIMS code base, but I'm guessing these are meant to be user-facing methods. Just looking for confirmation on this. 

I've included comments for each.

jillianderson8 committed:
Improve emissions and quantity typing and aggregation logic

- add shared Number alias and type annotations for Emissions and quantity helpers
- simplify Emissions constructor and arithmetic, returning new typed instances
- rework emissions aggregation methods with clearer totals and docstrings
- tighten ProvidedQuantity/RequestedQuantity/DistributedSupply APIs and totals calculations

jillianderson8 committed:
Merge branch 'staging' into 537_quantity_aggregation
